extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -65,11 +68,36 @@ export class EncryptedMsgMailFormatter extends BaseMailFormatter {
     return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: emailIntroAndLinkBody, atts, isDraft: this.isDraft });
   }
 
-  private sendableSimpleTextMsg = async (newMsg: NewMsgData, pubs: PubkeyResult[], signingPrv?: Key) => {
-    const atts = this.isDraft ? [] : await this.view.attsModule.attach.collectEncryptAtts(pubs);
-    const { data: encryptedBody, type } = await this.encryptDataArmor(Buf.fromUtfStr(newMsg.plaintext), undefined, pubs, signingPrv);
-    const mimeType = type === 'smime' ? 'smimeEncrypted' : undefined;
-    return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: { "encrypted/buf": Buf.fromUint8(encryptedBody) }, type: mimeType, atts, isDraft: this.isDraft });
+  private sendableSimpleTextMsg = async (newMsg: NewMsgData, pubs: PubkeyResult[], signingPrv?: Key): Promise<SendableMsg> => {
+    console.log('-- Inside sendableSimpleTextMsg --');
+    const x509Pubs = pubs.filter(pub => pub.pubkey.type === 'x509').map(pub => pub.pubkey);
+    if (x509Pubs) {
+      // smime case
+      const atts: Att[] = this.isDraft ? [] : await this.view.attsModule.attach.collectAtts(); // collects attachments
+      const msgBody = this.richtext ? { 'text/plain': newMsg.plaintext, 'text/html': newMsg.plainhtml } : { 'text/plain': newMsg.plaintext };
+      const mimeEncodedPlainMessage = await Mime.encode(msgBody, { Subject: newMsg.subject }, atts);
+      const encryptedMessage = await SmimeKey.encryptMessage({ pubkeys: x509Pubs, data: Buf.fromUtfStr(mimeEncodedPlainMessage) });
+      console.log('-- End sendableSimpleTextMsg --');
+      /*
+            rootNode.setContent(body); // tslint:disable-line:no-unsafe-any
+            rootNode.addHeader('Content-Transfer-Encoding', 'base64'); // tslint:disable-line:no-unsafe-any
+            rootNode.addHeader('Content-Disposition', 'attachment; filename="smime.p7m"'); // tslint:disable-line:no-unsafe-any
+            rootNode.addHeader('Content-Description', 'S/MIME Encrypted Message'); // tslint:disable-line:no-unsafe-any
+      */
+      //const att = new Att({ data: Buf.fromUint8(encryptedMessage.data), name: 'smime.p7m', contentDescription: 'S/MIME Encrypted Message' });
+      return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: { "encrypted/buf": Buf.fromUint8(encryptedMessage.data) }, type: 'smimeEncrypted', atts: [], isDraft: this.isDraft });
```

Comment:
Would be good to eventually have a specialized method like `createSmimeEncrypted` that would not have all the noise